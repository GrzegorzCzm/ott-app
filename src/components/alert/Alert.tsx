import React from "react";
import { useDispatch, useSelector } from "react-redux";

import Snackbar from "@material-ui/core/Snackbar";
import IconButton from "@material-ui/core/IconButton";
import CloseIcon from "@material-ui/icons/Close";

import { closeAlertAction } from "../../redux/actions/alert";

const Alert = () => {
  const dispatch = useDispatch();
  const { isAlert, alertText } = useSelector((state: any) => state.alerts);

  const closeAlert = () => {
    dispatch(closeAlertAction());
  };

  return (
    <Snackbar
      autoHideDuration={6000}
      anchorOrigin={{
        vertical: "bottom",
        horizontal: "left",
      }}
      open={isAlert}
      onClose={closeAlert}
      message={alertText}
      action={
        <IconButton
          size="small"
          aria-label="close"
          color="inherit"
          onClick={closeAlert}
        >
          <CloseIcon fontSize="small" />
        </IconButton>
      }
    />
  );
};

export default Alert;
