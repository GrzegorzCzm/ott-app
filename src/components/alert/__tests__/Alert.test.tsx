import React from "react";
import * as redux from "react-redux";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import Alert from "../Alert";

describe("Alert", () => {
  const spySelector = jest.spyOn(redux, "useSelector");
  const spyDispatch = jest.spyOn(redux, "useDispatch");
  const mockDispatch = jest.fn();

  spyDispatch.mockReturnValue(mockDispatch);

  it("renders correctly when isAlert is false", () => {
    spySelector.mockReturnValue({
      isAlert: false,
      alertText: "",
    });

    const wrapper = shallow(<Alert />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders correctly when text is set and isAlert is true", () => {
    const SOME_TEXT = "some text";
    spySelector.mockReturnValue({
      isAlert: true,
      alertText: SOME_TEXT,
    });

    const wrapper = shallow(<Alert />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
