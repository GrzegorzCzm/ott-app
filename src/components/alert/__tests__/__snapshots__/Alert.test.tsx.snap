// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert renders correctly when isAlert is false 1`] = `
<WithStyles(ForwardRef(Snackbar))
  action={
    <ForwardRef(WithStyles)
      aria-label="close"
      color="inherit"
      onClick={[Function]}
      size="small"
    >
      <UNDEFINED
        fontSize="small"
      />
    </ForwardRef(WithStyles)>
  }
  anchorOrigin={
    Object {
      "horizontal": "left",
      "vertical": "bottom",
    }
  }
  autoHideDuration={15000}
  message=""
  onClose={[Function]}
  open={false}
/>
`;

exports[`Alert renders correctly when text is set and isAlert is true 1`] = `
<WithStyles(ForwardRef(Snackbar))
  action={
    <ForwardRef(WithStyles)
      aria-label="close"
      color="inherit"
      onClick={[Function]}
      size="small"
    >
      <UNDEFINED
        fontSize="small"
      />
    </ForwardRef(WithStyles)>
  }
  anchorOrigin={
    Object {
      "horizontal": "left",
      "vertical": "bottom",
    }
  }
  autoHideDuration={15000}
  message="some text"
  onClose={[Function]}
  open={true}
/>
`;
